tensorflow_estimator/python/estimator/exporter.py

```diff
@@ -292,10 +293,11 @@ def export(self, estimator, export_path, checkpoint_path, eval_result,
                                              self._event_file_pattern)
       self._best_eval_result = self._get_best_eval_result(
           full_event_file_pattern)
-
+    
     if self._best_eval_result is None or self._compare_fn(
         best_eval_result=self._best_eval_result,
-        current_eval_result=eval_result):
+        current_eval_result=eval_result) or self._is_first_export:
```

Comment:
This shouldn't be necessary-- best_eval_result should be None at the beginning, correct? Can you clarify what this is fixing?

Reply:
For example, we do evaluation every 10000 steps. If the best metrics appear in the first evaluation, there will be nothing in the best exporter directory.

Reply:
Okay, got it. Can we:

1. Change the flag to _has_exported (or something better) that is False by default, and flip it to true instead?
2. Add tests.
3. Add some comments explaining what the flag is for and this scenario that it catches?

Thanks--